File: project1/link.py
class stuff:
    def __init__(self,num = None,nxt = None):
        self.num = num
        self.nxt = nxt

ship = [stuff(i,None) for i in range(30)]
for i in range(29):
    ship[i].nxt = ship[i+1]
ship[29].nxt = ship[0]

ans = []
total = 0
cur = ship[0]
pre = ship[29]

for i in range(15):
    cnt = 0
    for j in range(8):
        cur = cur.nxt
        pre = pre.nxt
    ans.append(cur.num)
    pre.nxt = cur.nxt
    tmp = cur
    del(tmp)
    cur = cur.nxt
print ans

File: project1/link.c
#include <stdio.h>
#include <stdlib.h>
typedef struct stuff{
    struct stuff* nxt;
    char num;
}stuff;

int main(){
    //initial;
    stuff* ship[30];
    for(int i = 0; i<30; i++) ship[i] = (stuff*)malloc(sizeof(stuff));
    for(int i = 0; i<30; i++){
        ship[i]->num = i;
        if(i<29){
            ship[i]->nxt = ship[i+1];
        }
        else{
            ship[i]->nxt = ship[0];
        }
    }

    //core;
    int ans[15];
    stuff* cur = ship[0];
    stuff* pre = ship[29];
    for(int cnt = 0; cnt<15; cnt++){
        for(int i = 0; i<8; i++){
            cur = cur->nxt;
            pre = pre->nxt;
        }
        ans[cnt] = cur->num;
        pre->nxt = cur->nxt;
        stuff* tmp = cur;
        free(tmp);
        cur = pre->nxt;
    }
    for(int i = 0; i<15; i++){
        printf("%d ",ans[i]);
    }printf("\n");
    return 0;
}
